test/tests_multiple_instances.js

```diff
@@ -0,0 +1,126 @@
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], 'Processing request');
+        var params = JSON.parse(message[1]);
+
+        test.assertEquals(params.app_key, "YOUR_APP_KEY1");
+        test.assert(exists(params.device_id));
+        test.assert(exists(params.timestamp));
+        test.assert(exists(params.hour));
+        test.assert(exists(params.dow));
+        
+        params.events = JSON.parse(params.events);
+        test.assertEquals(params.events.length, 1);
+        
+        test.assertEquals(params.events[0].key, 'first_app');
+        test.assertEquals(params.events[0].count, 1);
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], 'Sending XML HTTP request');
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY2] Processing request');
+        var params = JSON.parse(message[1]);
+
+        test.assertEquals(params.app_key, "YOUR_APP_KEY2");
+        test.assert(exists(params.device_id));
+        test.assert(exists(params.timestamp));
+        test.assert(exists(params.hour));
+        test.assert(exists(params.dow));
+        
+        params.events = JSON.parse(params.events);
+        test.assertEquals(params.events.length, 1);
+        
+        test.assertEquals(params.events[0].key, 'second_app');
+        test.assertEquals(params.events[0].count, 1);
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY2] Sending XML HTTP request');
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY3] Processing request');
+        var params = JSON.parse(message[1]);
+
+        test.assertEquals(params.app_key, "YOUR_APP_KEY3");
+        test.assert(exists(params.device_id));
+        test.assert(exists(params.timestamp));
+        test.assert(exists(params.hour));
+        test.assert(exists(params.dow));
+        
+        params.events = JSON.parse(params.events);
+        test.assertEquals(params.events.length, 1);
+        
+        test.assertEquals(params.events[0].key, 'third_app');
+        test.assertEquals(params.events[0].count, 1);
+    });
+    tests.push(function (message){
+        test.assertEquals(message[0], '[YOUR_APP_KEY3] Sending XML HTTP request');
+    });
+    casper.removeAllListeners('remote.message');
+    casper.on('remote.message', function(message) {
+        this.echo(message);
+        if (tests[cnt]) {
+            tests[cnt](message.split("\n"));
+            cnt++;
+        }
+    });
+    casper.start(fs.workingDirectory+"/examples/example_multiple_instances.html", function() {
+    }).run(function() {
+        setTimeout(function(){
+            casper.clear();
+            casper.clearCache();
+            casper.clearMemoryCache();
+            casper.removeAllListeners('remote.message');
+            casper.open(fs.workingDirectory+"/test/files/clear.html", function() {});
```

Comment:
Codacy found an issue: ['casper' is not defined.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6356797)